heareval/baseline.py

```diff
@@ -58,50 +53,63 @@ def forward(self, x: Tensor):
         # Convert to a log mel spectrum.
         x = torch.log(x + self.epsilon)
 
-        # Apply projections to get all required embeddings
-        x4096 = x.matmul(self.emb4096)
-        x2048 = x.matmul(self.emb2048)
-        x512 = x.matmul(self.emb512)
-        x128 = x.matmul(self.emb128)
-        x20 = x.matmul(self.emb20)
-
-        # The 20-dimensional embedding is specified to be int8. To cast to int8 we'll
-        # apply an activation to ensure the embedding is in a 0 to 1 range first.
-        x20 = self.activation(x20)
-
-        # Scale to int8 value range and cast to int
-        int8_max = torch.iinfo(torch.int8).max
-        int8_min = torch.iinfo(torch.int8).min
-        x20 = x20 * (int8_max - int8_min) + int8_min
-        x20 = x20.type(torch.int8)
-
-        return {4096: x4096, 2048: x2048, 512: x512, 128: x128, 20: x20}
-
+        # Apply projection to get a 4096 dimension embedding
+        embedding = x.matmul(self.projection)
 
-def input_sample_rate() -> int:
-    """
-    Returns:
-        One of the following values: [16000, 22050, 44100, 48000].
-            To avoid resampling on-the-fly, we will query your model
-            to find out what sample rate audio to provide it.
-    """
-    return RandomProjectionMelEmbedding.sample_rate
+        return embedding
 
 
-def load_model(model_file_path: str, device: str = "cpu") -> Any:
+def load_model(
+    model_file_path: str = "basic", device: str = "cpu"
+) -> Tuple[torch.nn.Module, Dict[str, Any]]:
     """
     In this baseline, we don't load anything from disk.
 
     Args:
-        model_file_path: Load model checkpoint from this file path.
-            device: For inference on machines with multiple GPUs,
+        model_file_path: Load model checkpoint from this file path. For this baseline
+            there is a basic version that contains no learning. To use the basic model
+            pass in 'basic' as the path. To load a random-init version of the learned
+            baseline pass in 'learned'. Passing in the location of a saved learned
+            baseline model will attempt to load that from disk.
```

Comment:
Hmm this still seems kinda gross to me. Because you can do learning on the "basic" model and train the random projection to become a learned projection. This also kinda sets a bad example for how the API should be used. Should be reconsider and put everything in the class?

Reply:
Then we can subclass from FrameBasedEmbeddingModel for framing?

Reply:
Alternately, we could just stick to the one module per model idea. And have heareval.baseline.linear and heareval.baseline.convnet